resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CallNotPermittedException.java

```diff
@@ -24,8 +24,11 @@
  */
 public class CallNotPermittedException extends RuntimeException {
 
-    private CallNotPermittedException(String message, boolean writableStackTrace) {
+    private final transient CircuitBreaker circuitBreaker;
+
+    private CallNotPermittedException(CircuitBreaker circuitBreaker, String message, boolean writableStackTrace) {
         super(message, null, false, writableStackTrace);
+        this.circuitBreaker = circuitBreaker;
```

Comment:
I'm not sure if we should have a reference in every exception to the CircuitBreaker instance.
I would prefer if we just add the name to the CallNotPermittedException as a field.

Reply:
I also felt a bit uncomfortable doing this, but also didn't want future requests asking to add other fields in the exception. What's the major concern with adding a reference?

Reply:
Any update? Can change to just having string if there are major concerns.

Reply:
Yes, please change to to a string.

Reply:
Done.